src/v/cloud_storage/cache_service.cc

```diff
@@ -418,7 +436,16 @@ ss::future<> cache::trim(
         co_return;
     }
 
-    auto [walked_cache_size, filtered_out_files, candidates_for_deletion, _]
+    // We are going to do a walk, rearm the periodic tracker sync if it is about
+    // to run soon.
+    _tracker_sync_timer.rearm(ss::lowres_clock::now() + tracker_sync_period);
```

Comment:
Why do we need to rearm timer here?
The method is called only from `start` method of the cache and after the call to `clean_up_at_start` the timer is armed. So it looks like at this point the timer shouldn't be armed.